app/src/main/java/org/onionshare/android/ShareManager.kt

```diff
@@ -158,7 +160,8 @@ class ShareManager @Inject constructor(
         }
         startSharingJob = null
 
-        if (torManager.state.value !is TorState.Stopped) torManager.stop()
+        val torState = torManager.state.value
+        if (torState !is TorState.Stopped && torState !is TorState.Stopping) torManager.stop()
```

Comment:
Unfortunately I think there's still a race here, because this happens on Dispatchers.IO. There could be two concurrent calls to this method (eg from the UI and from stopOnError()), both of which would check the current state and find that it's Starting or Published, and thus both would call stop().

Reply:
I wonder if there's a similar issue with the web server, or whether it ignores redundant calls to stop()?

Reply:
I added a commit to fix the tor race. Last I checked, the webserver tolerated repeated calls to stop().

Reply:
For me, it would be fine to disregard this unlikely scenario for now. The ui prevents button presses when in Stopping and only releases the button once stopped.